controllers/repo_manager/controller.go

```diff
@@ -321,23 +325,26 @@ func (r *RepoManagerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	pulpController, err = r.pulpStatus(ctx, pulp, log)
 	if err != nil {
 		return pulpController, err
+	} else if pulpController.RequeueAfter > 0 {
+		return pulpController, nil
 	} else if pulpController.Requeue {
 		return pulpController, nil
 	}
 
 	// If we get into here it means that there is no reconciliation
 	// nor controller tasks pending
 	log.Info("Operator tasks synced")
-	return pulpController, nil
+	return ctrl.Result{}, nil
 }
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *RepoManagerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	// creates a new eventRecorder to be able to interact with events
 	r.recorder = mgr.GetEventRecorderFor("Pulp")
+
 	controller := ctrl.NewControllerManagedBy(mgr).
-		For(&repomanagerv1alpha1.Pulp{}).
+		For(&repomanagerv1alpha1.Pulp{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
```

Comment:
this predicate will avoid triggering a reconciliation only for `pulp.status`
we need to set this way instead of

```
For(&repomanagerv1alpha1.Pulp{}).
WithPredicates(predicate.GenerationChangedPredicate{})).
Owns(&appsv1.StatefulSet{}).
Owns(&appsv1.Deployment{}).
...
```
because the second option would ignore the status field update from all resources managed by the operator, which is something we dont want, it would break things like
https://github.com/pulp/pulp-operator/blob/6df43eecbd501a79feba6d002b40767384de9f3b/controllers/repo_manager/status.go#L212